### PR TITLE
Implementation of ACTIVITY_SIGNALED event

### PR DESCRIPTION
**Pull request by CTI777**
Related to https://github.com/Activiti/Activiti/issues/2272
Depends on 
https://github.com/Activiti/activiti-api/pull/84
https://github.com/Activiti/activiti-cloud-api/pull/65
https://github.com/Activiti/Activiti/pull/2511

### CODE REVIEW

File: activiti-cloud-starter-runtime-bundle/src/test/java/org/activiti/cloud/starter/tests/services/audit/AuditProducerIT.java [hunk 1]
```diff
@@ -545,6 +551,102 @@ public void shouldEmitEventsForTaskAddDeleteGroupCandidates() {
...
+                                                                    });
+
+        await("Broadcast Signals").untilAsserted(() -> {
+            assertThat(streamHandler.getReceivedHeaders()).containsKeys(RUNTIME_BUNDLE_INFO_HEADERS);
+
+            List<CloudRuntimeEvent<?, ?>> receivedEvents = streamHandler.getLatestReceivedEvents();
+
+            assertThat(streamHandler.getReceivedHeaders()).containsKeys(ALL_REQUIRED_HEADERS);
+
+            assertThat(receivedEvents)
```
Comment (erdemedeiros) [inline activiti-cloud-starter-runtime-bundle/src/test/java/org/activiti/cloud/starter/tests/services/audit/AuditProducerIT.java]: Can you add another assert to check that `processDefinitionId` and `processInstanceId` are set properly?

**Comment by CTI777** [comment]
@salaboy @erdemedeiros @igdianov 
It seems that we have a problem with MessageProducerCommandContextCloseListener.
In the test when two processes are listening for one signal a processInstanceId in the events related to second process is overwritten with processInstanceId  from the first process..
This bug should be fixed, maybe inside this PR? Details should be considered (discussed) additionally, because the header for the messages' bunch should be modified... And this may lead to failing of some other tests...

**Comment by salaboy** [comment]
@CTI777 hmm that doesn't sounds like the right behaviour.. we will need to investigate.. Please sync with @erdemedeiros about this.

**Comment by igdianov** [comment]
@Salaboy the close listener injects the execution attributes into events before sending it to the stream  under the assumption that there is always single process instance emitting events inside each transaction. This is apparently not the case, so it should be the responsibility of the event aggregator to resolve and apply correct values to each event when it is added to the list. Once the event is added to the list, it would be good practice to make it immutable. However, this is currently not possible because the setters are public. We could possibly wrap it into delegate to seal the event once we add it to the list. 

I will help @cti777 to fix this issue.

**Comment by erdemedeiros** [comment]
@igdianov on the same subject: I noticed that currently `ProcessDeployedEvent`s are not setting headers at all (as they are not using `closeListener`). I created a specific `MessageBuilderChainFactory` to set only routing key and runtime bundle info headers [in this PR](https://github.com/Activiti/activiti-cloud-runtime-bundle-service/pull/251/files#diff-ffe0cdb270889fdd18b150863aed616bR30) . However it's not implementing the interface `MessageBuilderChainFactory` as I wasn't sure which information I could use add more information there (if there is any). This new `MessageBuilderChainFactory` is invoked in the [CloudProcessDeployedProducer](https://github.com/Activiti/activiti-cloud-runtime-bundle-service/pull/251/files#diff-a0147315b7913b5b6f481ccb7ee94e31R44).

**Comment by igdianov** [comment]
> @igdianov on the same subject: I noticed that currently `ProcessDeployedEvent`s are not setting headers at all (as they are not using `closeListener`). I created a specific `MessageBuilderChainFactory` to set only routing key and runtime bundle info headers [in this PR](https://github.com/Activiti/activiti-cloud-runtime-bundle-service/pull/251/files#diff-ffe0cdb270889fdd18b150863aed616bR30) . However it's not implementing the interface `MessageBuilderChainFactory` as I wasn't sure which information I could use add more information there (if there is any). This new `MessageBuilderChainFactory` is invoked in the [CloudProcessDeployedProducer](https://github.com/Activiti/activiti-cloud-runtime-bundle-service/pull/251/files#diff-a0147315b7913b5b6f481ccb7ee94e31R44).

@erdemedeiros I will take a look at it.

**Comment by igdianov** [comment]
@CTI777 @erdemedeiros This PR needs to be rebased on top of develop branch. We also need to add some tests to make sure that execution context is injected in signal audit events.

**Comment by igdianov** [comment]
@erdemedeiros I have fixed missing execution context attributes for CloudBPMNSignalEvent. See test.

There is a small nuisance with start event catch signal. The event is dispatched outside of transaction, so it's entity does not have associated process instance id to retrieve the details of current execution context.

**Pull request merged**